crates/ra_hir_ty/src/tests.rs

```diff
@@ -37,6 +37,14 @@ use crate::{
 // against snapshots of the expected results using insta. Use cargo-insta to
 // update the snapshots.
 
+fn with_tracing_logs<T>(action: impl FnOnce() -> T) -> T {
+    use tracing_subscriber::{layer::SubscriberExt, EnvFilter, Registry};
+    use tracing_tree::HierarchicalLayer;
+    let filter = EnvFilter::from_env("CHALK_DEBUG");
+    let subscriber = Registry::default().with(filter).with(HierarchicalLayer::new(2));
```

Comment:
If you'd like, you can try rendering the output with ASCII lines: https://docs.rs/tracing-tree/0.1.2/tracing_tree/struct.HierarchicalLayer.html#method.with_indent_lines

Reply:
I tried it, but I like it a bit better without the lines...